src/vivarium_public_health/population/data_transformations.py

```diff
@@ -8,67 +8,94 @@
 
 """
 from collections import namedtuple
+from typing import Tuple, Union
 
 import numpy as np
 import pandas as pd
+from vivarium.framework.randomness import RandomnessStream
 
+_SORT_ORDER = ["location", "year_start", "sex", "age_start"]
 
-def assign_demographic_proportions(population_data):
-    """Calculates conditional probabilities on the provided population data for use in sampling.
+
+def assign_demographic_proportions(
+    population_data: pd.DataFrame,
+    include_sex: str,
+) -> pd.DataFrame:
+    """Calculates conditional probabilities on the provided population data for sampling.
 
     Parameters
     ----------
-    population_data : pandas.DataFrame
-        Table with columns 'age', 'sex', 'year', 'location', and 'population'
+    population_data
+        Table with columns 'age', 'sex', 'year', 'location', and 'value'
+    include_sex
+        'Female', 'Male', or 'Both'.  Sexes to include in the distribution.
 
     Returns
     -------
     pandas.DataFrame
-        Table with the same columns as `population_data` and additionally with columns
-        'P(sex, location, age| year)', 'P(sex, location | age, year)', and
-        'P(age | year, sex, location)' with values calculated from the
-        various population levels.
+        Table with columns
+            'age' : Midpoint of the age group,
+            'age_start' : Lower bound of the age group,
+            'age_end' : Upper bound of the age group,
+            'sex' : 'Male' or 'Female',
+            'location' : location,
+            'year' : Year,
+            'population' : Total population estimate,
+            'P(sex, location | age, year)' : Conditional probability of sex and
+            location given age and year,
+            'P(sex, location, age | year)' : Conditional probability of sex, location,
+            and age given year,
+            'P(age | year, sex, location)' : Conditional probability of age given
+            year, sex, and location.
     """
+    population_data = population_data.copy()
+    if include_sex != "Both":
```

Comment:
This is the core bit that updates the probability distributions we use for sampling.